### DISCUSSION TITLE
Can RMLUI load back buffer textures?

### DISCUSSION THREAD

**Discussion by moonshineTheleocat** [mikke89/RmlUi#285] [Q&A]
For example. If I have a model constantly being rendered in a back buffer, and want the texture to be accessed by RMLui, can RMLUI do that? And if so how?

Or am I required to download the texture from the GPU and reupload it for the Library?

**Comment by mikke89** [comment]
Hi.

RmlUi does not touch your graphics API, it communicates everything it wants to do through its render interface, and then you as the client render the commands however you like. This means that you can easily render your own textures and they can stay entirely on the client side.

So for example, you could designate a special texture name for you back buffer. When RmlUi asks you to load this you provide a reference handle for it. Then, at some point RmlUi will you hand you that handle back and ask you to render it. Then it is just a matter of drawing the texture you have the way you like.

Instead of a special texture name you could also create a custom decorator or even a custom element to achieve this.
